Project/Source/lcdfunctions.c

```diff
@@ -0,0 +1,264 @@
+		if(player1->positionY+yt <= yScreenEnd && player1->positionY+yt >= yScreenStart)	
+			y=player1->positionY+yt;																																		
+		LCD_DrawFullCircle(x, y, r);
+		LCD_FillTriangle(x+a/2,x-a/2,x,y+r,y+r,y+r+a);
+	}
+	else if(player1->direction==2){
+		if(player1->positionX-xt <= xScreenEnd && player1->positionX-xt >= xScreenStart)
+			x=player1->positionX-xt;	
+		LCD_DrawFullCircle(x, y, r);
+		LCD_FillTriangle(x-r,x-r,x-r-a,y+a/2,y-a/2,y);
+	}
+	else if(player1->direction==3){
+		if(player1->positionY-yt <= yScreenEnd && player1->positionY-yt >= yScreenStart)
+			y=player1->positionY-yt;
+		LCD_DrawFullCircle(x, y, r);
+		LCD_FillTriangle(x-a/2,x+a/2,x,y-r,y-r,y-r-a);
+	}
+	else{
+		if(player1->positionX+xt <= xScreenEnd && player1->positionX+xt >= xScreenStart)
+			x=player1->positionX+xt;	
+		LCD_DrawFullCircle(x, y, r);
+		LCD_FillTriangle(x+r,x+r,x+r+a,y-a/2,y+a/2,y);
+	}
+	
+	player1->positionX = x;
+	player1->positionY = y;
+}
+
+void TankRemove(PlayerPosition player){			/* brisanje tenka */
+	LCD_SetTextColor(LCD_COLOR_WHITE);		/* prima poziciju tenka */
+	int x,y;
+	x=player.positionX;	
+	y=player.positionY;
+	if(player.direction==1){
+		LCD_DrawFullCircle(x, y, r);
+		LCD_FillTriangle(x+a/2,x-a/2,x,y+r,y+r,y+r+a);
+	}
+	else if(player.direction==2){
+		LCD_DrawFullCircle(x, y, r);
+		LCD_FillTriangle(x-r,x-r,x-r-a,y+a/2,y-a/2,y);
+	}
+	else if(player.direction==3){
+		LCD_DrawFullCircle(x, y, r);
+		LCD_FillTriangle(x-a/2,x+a/2,x,y-r,y-r,y-r-a);
+	}
+	else{
+		LCD_DrawFullCircle(x, y, r);
+		LCD_FillTriangle(x+r,x+r,x+r+a,y-a/2,y+a/2,y);
+	}
+}
+
+void TankRotate(PlayerPosition *player1, int next, int choosePlayer){    /* rotacija tenkova */
+	int x,y;
+	
+	x=player1->positionX;	
+	y=player1->positionY;
+	
+	TankRemove(*player1);
+	if(choosePlayer == 0)
+		LCD_SetTextColor(LCD_COLOR_BLUE);
+	else
+		LCD_SetTextColor(LCD_COLOR_RED);
+	
+	if(player1->direction==1){
+		if(next==0){
+			LCD_DrawFullCircle(x, y, r);
+			LCD_FillTriangle(x+r,x+r,x+r+a,y-a/2,y+a/2,y);
+		}
+		else{
+			LCD_DrawFullCircle(x, y, r);
+			LCD_FillTriangle(x-r,x-r,x-r-a,y+a/2,y-a/2,y);   /* npr. smjer=2, next=0 => rotira u 1; next=1 =>rotira u 3 */
+		}
+	}
+	else if(player1->direction==2){
+		if(next==0){
+			LCD_DrawFullCircle(x, y, r);
+			LCD_FillTriangle(x+a/2,x-a/2,x,y+r,y+r,y+r+a);
+		}
+		else{
+			LCD_DrawFullCircle(x, y, r);
+			LCD_FillTriangle(x-a/2,x+a/2,x,y-r,y-r,y-r-a);
+		}
+	}
+	else if(player1->direction==3){
+		if(next==0){
+			LCD_DrawFullCircle(x, y, r);
+			LCD_FillTriangle(x-r,x-r,x-r-a,y+a/2,y-a/2,y);
+		}
+		else{
+			LCD_DrawFullCircle(x, y, r);
+			LCD_FillTriangle(x+r,x+r,x+r+a,y-a/2,y+a/2,y);
+		}
+	}
+	else{
+		if(next==0){
+			LCD_DrawFullCircle(x, y, r);
+			LCD_FillTriangle(x-a/2,x+a/2,x,y-r,y-r,y-r-a);
+		}
+		else{
+			LCD_DrawFullCircle(x, y, r);
+			LCD_FillTriangle(x+a/2,x-a/2,x,y+r,y+r,y+r+a);
+		}
+	}
+	
+	player1->direction+= next == 0 ? (player1->direction==1 ? 3 : -1) : (player1->direction==4 ? -3 : 1);
```

Comment:
moze objasnjenje ovog uvjeta na liniji 251?

Reply:
Varijabla next sadrzi smjer rotacija (0 -> lijevo, 1 -> desno). Ako je next=0 postojecoj orijentaciji pribrajamo -1 (to je ekvivalentno okretanju za 90 u lijevo), ako je next=1 pribrajamo 1. Osim toga treba pripazit i na edge case ako je trenutna rotacija 4 ili 1 tada ne mozemo samo pribrojat 1/-1 nego mora ici iz 4 u 1 i obrnuto. (moguće orijentacije igrača su 1,2,3,4).

Reply:
ok.
ja se ovog nikad nebi sjetio,samo bi puknuo negdje gore da taj direction bude jos %4

ali..ako radi,dobro je :)

Reply:
Nažalost modulo ne pomaže za donji edge case tako ćemo se morat zadovoljit s ovim :)